### PR TITLE
Update assets page layout

### PR DESCRIPTION
**Pull request by ayumitk**
**Pull Request Summary**

ref: https://github.com/AstarNetwork/astar-apps/pull/922

**Check list**

- [ ] contains breaking changes
- [ ] adds new feature
- [x] modifies existing feature (bug fix or improvements)
- [ ] relies on other tasks
- [ ] documentation changes
- [ ] tested on mobile devices

**This pull request makes the following changes:**

**Changes**

- (ex: Change document B)

**To-dos**

- [x]  Basic layout and style update
- [x]  Reward animation
- [x] Use i18n
- [x]  Merge account unification
- [ ] Add to/remove from favourite button

### CODE REVIEW

**Comment by ayumitk** [comment]
268000190-077600b6-2007-4e2d-8eb5-93ae24b42941

**Comment by Kahonnohak** [comment]
Thanks Ayumi!

- Would it be possible to align all token amounts including ASTR? no preference of where but aligning with ASTR balance might make sense
- Could you also align the right side icons please?
- Also shall we use "Send" icon for ASTR transfer?
Screenshot 2023-10-05 at 12 00 00

- I have added a new colour #141E3E directly written sorry... later please add as Navy-4

- Some texts are default Black but please change them to Navy-1

- When it has the mobile header shall we use the mobile account style?
Screenshot 2023-10-05 at 12 03 14

- Now really want to change some icons BUT lets move on... I think after those changes we can invite Roy and Bobo to review this. There are funny behaviours at this moment but lets ask them to fix those. You could take a look at Reward animation.

Thanks

**Comment by ayumitk** [comment]
@Kahonnohak 

> Would it be possible to align all token amounts including ASTR? no preference of where but aligning with ASTR balance might make sense

I aligned all token positions by the right side of symbols, but do you think style A or B would be better?

|Current|A|B|
|-|-|-|
|assets-current|align-assets-a|align-assets-b|

**Comment by ayumitk** [comment]
@Kahonnohak When the balance is 0, I need a faucet icon now. So I searched for faucet in [Flaticon](https://www.flaticon.com/search?word=faucet&color=black&shape=outline) (image attached), but I think these are too complicated. Do you have any idea about it?

Screenshot 2023-10-05 at 12 52 55 PM

Screenshot 2023-10-05 at 12 57 28 PM

**Comment by Kahonnohak** [comment]
THANK YOU! Yes I meant A, is much easier to see numbers, although showing desimal are different each tokens... still A is good. Thank you so much.

And faucet..! forgotten about it. Let's go with this:
https://fonts.google.com/icons?selected=Material+Symbols+Rounded:water_drop:FILL@0;wght@300;GRAD@0;opsz@24&icon.query=water&icon.set=Material+Symbols&icon.style=Rounded
Screenshot 2023-10-05 at 21 39 29

**Comment by ayumitk** [comment]
@Kahonnohak I adjusted all of the above and added an animation for the Reward circle. Please review it

**Comment by Kahonnohak** [comment]
Nice work!! Lets review at creative sync later x

**Comment by Kahonnohak** [comment]
Small favour for now sorry, but can we see background fixed at top version? what do you think?

**Comment by impelcrypto** [comment]
@ayumitk 

Unify account
* Maybe implement it after merging [this](https://github.com/AstarNetwork/astar-apps/pull/925) PR? 

Add to/remove from favorite
* Maybe you store the data in localstorage like I did for `txHistories`. So the data object includes `address { network: { favoriteCoin } }`  Please check [this](https://github.com/AstarNetwork/astar-apps/blob/main/src/config/localStorage.ts#L25) file.

image

**Comment by ayumitk** [comment]
@impelcrypto Ah localstorage! thank you

**Comment by Kahonnohak** [comment]
@ayumitk thanks a lot! could you quickly fix one thing - I cannot scroll down the mobile nav for this moment. could you take a look?

**Comment by Kahonnohak** [comment]
@ayumitk all small things let me know if you don't have time

- Mobile header under line -> dark colour 1px
- Mobile evm network toggle -> move to the right
- Account Address -> font weight 700 and could you add network gradients if possible?
- Rewards numbers -> ASTR bottom like looks a bit below, small adjustment would be great
- Rewards animation -> A little slower - I think 16s look nice and peaceful
- Mobile padding in the asset list 32px
- Mobile assets search box -> please adjust - at least 16px apart?
- Dark colour asset list hover / locked token background colour -> 1F2F5F (Navy3)

Screenshot 2023-11-15 at 13 01 52

**Comment by impelcrypto** [comment]
@ayumitk 

> Could you add network change on this toggle please? Astar EVM(L1) <-> Astar zkEVM

You can add the button like:
Astar: disable the `Astar zkEVM` button for now.
Shibuya: just use <a> tag and go to 'zkatana-testnet/assets'
zKatana: just use <a> tag and go to 'shibuya-testnet/assets'

> Please also add claim function on the button on Native and Astar EVM.

You can refer to [this](https://github.com/AstarNetwork/astar-apps/blob/main/src/components/dapp-staking/my-staking/MyRewards.vue#L39) code

**Comment by impelcrypto** [comment]
@ayumitk 
> You can add the button like:
Astar: disable the Astar zkEVM button for now.
Shibuya: just use a tag (link) and open 'zkatana-testnet/assets'
zKatana: just use a tag (link) and open 'shibuya-testnet/assets'

Maybe you can try like `providerEndpoints[endpointKey.ASTAR_ZKEVM].networkAlias` to get `zkatana-testnet/` value

**Comment by Kahonnohak** [comment]
All accounts works like this - I hope it makes sense. I will add testnet and Shiden img later.
Screenshot 2023-11-16 at 15 26 16

**Comment by ayumitk** [comment]
@Kahonnohak I've updated the following. Please review it 🙏

> I cannot scroll down the mobile nav for this moment. could you take a look?

> Could you add network change on this toggle please? Astar EVM(L1) <-> Astar zkEVM

> - Dark theme header under line -> 1px ?
> - Mobile evm network toggle -> move to the right
> - Account Address -> font weight 700 and could you add network gradients if possible?
> - Metamask icon -> squashed?
> - Rewards numbers -> ASTR line bottom looks not matching with the number, small adjustment would be great
> - Rewards animation -> A little slower - I think 16s look nice and peaceful
> - Mobile padding in the asset list 32px
> - Mobile assets search box -> please adjust - at least 16px apart?
> - Happy to hide the star feature for now
> - Dark colour asset list hover / locked token background colour -> 1F2F5F (Navy3)
> - Adding one more thing - wanted to change this top padding below the header. Seems 27px is a nice position.

**Comment by Kahonnohak** [comment]
Thank you so much x

**Comment by Kahonnohak** [comment]
@impelcrypto Please add network switch and claiming rewards logic

**Comment by impelcrypto** [comment]
@ayumitk 
It looks cursor here isn't appropriate. The cursor is 'not-allowed' but I can click and claim the rewrds.
https://gyazo.com/bcbab020fdf42d7aa9ebbb43ca2a95b3

**Comment by impelcrypto** [comment]
@ayumitk @Kahonnohak Is this PR still under 'Draft'?

**Comment by ayumitk** [comment]
@Kahonnohak I've added the network switch part, but I still need to fix the rewards section logic.

@impelcrypto I've set this PR for ready. Let me ask about the rewards section logic at our sync later.

**Comment by ayumitk** [comment]
@impelcrypto I created the rewards section by copying [this code](https://github.com/AstarNetwork/astar-apps/blob/main/src/components/dapp-staking/my-staking/MyRewards.vue). And I have 2 issues here.

1. The estimated reward amount will not show after refreshing the assets page. But once we go to the dapps staking page, the number will appear. So it seems we need to fetch/store/watch some data first on the assets page, but I don't know which one.
https://www.loom.com/share/f683109c077a475192e65791ea768f02?sid=a6d8810d-4260-48b5-a309-3bb4099f3e69

2. The disabling of the button condition will not disappear even if it's possible to claim. This might related to the first issue.

The disabling of the button condition will not disappear even if it's possible to claim.

**Comment by impelcrypto** [comment]
@ayumitk 
I can see the faucet button instead of transfer button here. You might want to check the condition for Shibuya. 
image

**Comment by Kahonnohak** [comment]
Last request before going production please!

@impelcrypto please if @ayumitk did not have time
1) When a user do not have staking, please say `Start staking` and navigate to dApp Staking page.
2) When user has staking but no rewards to be claimed, please show 0 ASTR - at this moment, portal seems to be trying to find number.
3) Mobile staking box is not fit in the width. please change.
4) Please change the network toggle font to 12px
5) Please add account background for testnets (Shibuya and zKatana) also Shiden. (accound_bg_testnet/shiden) I added to img folder already.

I have checked most browsers and android/iphone. All look great!!
Screenshot 2023-11-21 at 13 56 06
Screenshot 2023-11-21 at 13 54 09

**Comment by ayumitk** [comment]
> I can see the faucet button instead of transfer button here. You might want to check the condition for Shibuya.

@impelcrypto It seems there is the same bug on the production. Could you please fix this?

Screenshot 2023-11-22 at 12 52 22 AM

**Comment by ayumitk** [comment]
@impelcrypto could you take care of 1 and 2 please?

> 1. When a user do not have staking, please say Start staking and navigate to dApp Staking page.
> 2. When user has staking but no rewards to be claimed, please show 0 ASTR - at this moment, portal seems to be trying to find number.

@Kahonnohak I've fixed 3, 4 and 5!

> 3. Mobile staking box is not fit in the width. please change.
> 4. Please change the network toggle font to 12px
> 5. Please add account background for testnets (Shibuya and zKatana) also Shiden. (accound_bg_testnet/shiden) I added to img folder already.

**Comment by impelcrypto** [comment]
> It seems there is the same bug on the production. Could you please fix this?

Production is Ok because the transfer button shows after expanding it. How about adding the transfer button next to the faucet button if the portal connects Shibuya?

image
.

**Comment by Kahonnohak** [comment]
Thank you Ayumi!

Yes, it is only Shibuya that has faucet and transfer at the same time, and anyways this should be the place for all action buttons:)
Screenshot 2023-11-21 at 21 27 45

**Comment by ayumitk** [comment]
@impelcrypto @Kahonnohak I've added a transfer button for shibuya in [this](https://github.com/AstarNetwork/astar-apps/pull/956/commits/57d2f8421786e2291a169f98930377792e415579) commit.

**Comment by impelcrypto** [comment]
@ayumitk 
> @impelcrypto @Kahonnohak I've added a transfer button for shibuya in [this](https://github.com/AstarNetwork/astar-apps/pull/956/commits/57d2f8421786e2291a169f98930377792e415579) commit.

Looks good. I think the 'expend' icon is missing for mobile devices.
https://gyazo.com/3a46a2124127c3e2017bc993a88595e8

**Comment by Kahonnohak** [comment]
> @ayumitk
> 
> > @impelcrypto @Kahonnohak I've added a transfer button for shibuya in [this](https://github.com/AstarNetwork/astar-apps/pull/956/commits/57d2f8421786e2291a169f98930377792e415579) commit.
> 
> Looks good. I think the 'expend' icon is missing for mobile devices, or is this according to the figma design? https://gyazo.com/3a46a2124127c3e2017bc993a88595e8

Oh sorry I missed this and it looks like the locked balance is missing the vesting tokens

Screenshot 2023-11-22 at 09 33 50

**Comment by Kahonnohak** [comment]
I see something went wrong. I'll check everything now
Screenshot 2023-11-22 at 09 40 44

**Review by Kahonnohak — APPROVED** [review]
LGTM!! Thank you so much for the great work!

File: src/components/assets/Account.vue [hunk 1]
```diff
@@ -240,6 +273,27 @@ export default defineComponent({
...
+    const bg_img = {
+      native: require('/src/assets/img/account_bg_native.webp'),
+      shiden: require('/src/assets/img/account_bg_shiden.webp'),
+      testnet: require('/src/assets/img/account_bg_testnet.webp'),
+      zk: require('/src/assets/img/account_bg_zk.webp'),
+      testnet_zk: require('/src/assets/img/account_bg_testnet_zk.webp'),
+    };
+
+    const bg = computed<String>(() => {
+      if (currentNetworkIdx.value === 0) {
```
Comment (bobo-k2) [inline src/components/assets/Account.vue:285]: Use `endpointKey` enum instead numbers

File: src/components/assets/Assets.vue [hunk 2]
```diff
@@ -167,6 +156,20 @@ export default defineComponent({
       window.removeEventListener(event, handler);
     });
 
+    const isDarkTheme = computed<boolean>(() => store.getters['general/theme'] === 'DARK');
```
Comment (bobo-k2) [inline src/components/assets/Assets.vue:159]: We have this line in so many places in our codebase. What do you think moving the line to some hook? It doesn't need to be in this PR

File: src/components/assets/Erc20Currency.vue [hunk 3]
```diff
@@ -1,118 +1,145 @@
...
+            <button class="btn btn--icon" @click="handleDeleteStoredToken(token.address)">
+              <astar-icon-delete size="22" />
+            </button>
+            <span class="text--expand-menu">{{ $t('remove') }}</span>
+            <q-tooltip>
+              <span class="text--tooltip">{{ $t('remove') }}</span>
+            </q-tooltip>
+          </div>
+
+          <!-- <div>
```
Comment (bobo-k2) [inline src/components/assets/Erc20Currency.vue:122]: Delete if not needed

File: src/components/assets/EvmAssetList.vue [hunk 4]
```diff
@@ -97,6 +106,13 @@ export default defineComponent({
       search.value = event.target.value;
     };
 
+    const store = useStore();
+    const isDarkTheme = computed<boolean>(() => store.getters['general/theme'] === 'DARK');
```
Comment (bobo-k2) [inline src/components/assets/EvmAssetList.vue:110]: Already commented above

File: src/components/assets/EvmNativeToken.vue [hunk 5]
```diff
@@ -1,69 +1,115 @@
...
-              <button class="btn btn--sm">
-                {{ $t('assets.faucet') }}
-              </button>
-            </a>
-            <button
-              v-else-if="isFaucet"
-              class="btn btn--sm"
-              @click="handleModalFaucet({ isOpen: true })"
+              <path
+                d="M11.9 17.61C11.18 17.66 10.36 17.44 9.47999 16.95C8.62999 16.47 8.07999 15.61 7.86999 14.4C7.81999 14.16 7.71999 13.97 7.55999 13.83C7.19999 13.51 6.63999 13.54 6.31999 13.91C6.13999 14.12 6.07999 14.37 6.13999 14.68C6.45999 16.39 7.23999 17.63 8.45999 18.37C9.55999 19.04 10.64 19.37 11.69 19.37C11.79 19.37 11.89 19.37 11.99 19.37C12.24 19.35 12.44 19.25 12.6 19.09C12.77 18.92 12.85 18.72 12.85 18.48C12.85 18.21 12.76 17.99 12.57 17.82C12.39 17.66 12.16 17.59 11.89 17.61H11.9Z"
```
Comment (bobo-k2) [inline src/components/assets/EvmNativeToken.vue:70]: Does it make sense to create svg file instead hadrdcoding path in the component? If we move svg to a separate file we have a reusable component

File: src/components/header/TroubleHelp.vue [hunk 6]
```diff
@@ -1,16 +1,24 @@
...
...28 7.52901 6.03109 7.50559 5.8156C7.48217 5.60011 7.53232 5.39573 7.65602 5.20244C8.13165 4.4833 8.71987 3.93644 9.42068 3.56187C10.1215 3.18729 10.9085 3 11.7817 3C13.1761 3 14.3111 3.39919 15.1867 4.19757C16.0622 4.99593 16.5 6.03562 16.5 7.31663C16.5 7.99494 16.3547 8.61653 16.064 9.18139C15.7734 9.74627 15.2779 10.3541 14.5777 11.0048C13.9219 11.599 13.4745 12.0814 13.2355 12.4518C12.9965 12.8221 12.8578 13.2384 12.8194 13.7007C12.7809 13.9672 12.6704 14.1893 12.4879 14.3669C12.3053 14.5446 12.0855 14.6335 11.8285 14.6335C11.5715 14.6335 11.3517 14.5455 11.1691 14.3697C10.9866 14.1938 10.8953 13.9774 10.8953 13.7205C10.8953 13.0938 11.0385 12.5208 11.325 12.0015C11.6114 11.4823 12.091 10.9237 12.7636 10.3258C13.4818 9.69555 13.9682 9.16971 14.2228 8.74831C14.4775 8.32691 14.6048 7.86889 14.6048 7.37424ZM11.7817 20C11.3985 20 11.0685 19.8616 10.7917 19.5849C10.5149 19.3082 10.3765 18.9783 10.3765 18.5954C10.3765 18.2124 10.5149 17.8825 10.7917 17.6058C11.0685 17.3291 11.3985 17.1907 11.7817 17.1907C12.1648 17.1907 12.4948 17.3291 12.7716 17.6058C13.0485 17.8825 13.1869 18.2124 13.1869 18.5954C13.1869 18.9783 13.0485 19.3082 12.7716 19.5849C12.4948 19.8616 12.1648 20 11.7817 20Z"
```
Comment (bobo-k2) [inline src/components/header/TroubleHelp.vue:14]: svg path, see the comment above

**Review by impelcrypto — APPROVED** [review]
LGTM
@ayumitk Let's move some icons to Astar-UI later

**Review by bobo-k2 — CHANGES_REQUESTED** [review]

File: src/components/assets/Account.vue [hunk 7]
```diff
@@ -1,87 +1,118 @@
...
-                  <span class="text--tooltip">{{ $t(isH160 ? 'blockscout' : 'subscan') }}</span>
-                </q-tooltip>
-              </a>
+          <div class="row--actions">
+            <div v-if="isAccountUnification">
+              <button type="button" class="btn--icon" @click="showAccountUnificationModal()">
+                <astar-icon-person />
+              </button>
+              <q-tooltip>
+                <span class="text--tooltip">Unify accounts</span>
```
[the remark this answers is not in the document]
Comment (ayumitk) [reply src/components/assets/Account.vue]: Fixed it in this [commit](https://github.com/AstarNetwork/astar-apps/pull/956/commits/0b71f3a800b60d1bae2a14f328a21aaea62291e1)

File: src/components/assets/Account.vue [hunk 8]
```diff
@@ -1,87 +1,118 @@
...
-              <div>
-                <button id="copyAddress" type="button" class="icon--primary" @click="copyAddress">
-                  <astar-icon-copy />
-                </button>
-                <q-tooltip>
-                  <span class="text--tooltip">{{ $t('copy') }}</span>
-                </q-tooltip>
+              <div class="text--balance">
+                {{ $n(totalBal) }}
+                <span>USD</span>
```
[the remark this answers is not in the document]
Comment (ayumitk) [reply src/components/assets/Account.vue]: Fixed it in this [commit](https://github.com/AstarNetwork/astar-apps/pull/956/commits/0b71f3a800b60d1bae2a14f328a21aaea62291e1)

File: src/components/assets/Account.vue [hunk 9]
```diff
@@ -1,87 +1,118 @@
...
+                  Astar zkatana
+                </a>
+                <a v-else-if="currentNetworkIdx !== 1" class="btn" disabled>Astar zkEVM</a>
+              </template>
+            </template>
+
+            <!-- Native -->
+            <div v-else class="btn active">
+              {{ currentNetworkIdx === 4 ? 'Astar' : currentNetworkName.replace('Network', '') }}
+              Native
```
[the remark this answers is not in the document]
Comment (ayumitk) [reply src/components/assets/Account.vue]: Fixed it in this [commit](https://github.com/AstarNetwork/astar-apps/pull/956/commits/0b71f3a800b60d1bae2a14f328a21aaea62291e1)

File: src/components/assets/Account.vue [hunk 10]
```diff
@@ -1,87 +1,118 @@
...
+              </template>
+
+              <!-- Astar EVM -->
+              <template v-else>
+                <div class="btn active">
+                  {{ currentNetworkName.replace('Network', '') }}
+                  EVM (L1)
+                </div>
+                <a v-if="currentNetworkIdx === 2" class="btn" href="/zkatana-testnet/assets">
+                  Astar zkatana
```
[the remark this answers is not in the document]
Comment (ayumitk) [reply src/components/assets/Account.vue]: Fixed it in this [commit](https://github.com/AstarNetwork/astar-apps/pull/956/commits/0b71f3a800b60d1bae2a14f328a21aaea62291e1)

File: src/components/assets/Account.vue [hunk 11]
```diff
@@ -1,87 +1,118 @@
...
+        <div class="account-bg" :style="{ backgroundImage: `url(${bg})` }" />
+
+        <div class="wallet-tab">
+          <div class="wallet-tab__bg">
+            <!-- EVM -->
+            <template v-if="isH160">
+              <!-- zkEVM -->
+              <template v-if="isZkEvm">
+                <a class="btn" href="/shibuya-testnet/assets"> Shibuya EVM (L1) </a>
+                <div v-if="isZkEvm" class="btn active">Astar zkatana</div>
```
[the remark this answers is not in the document]
Comment (ayumitk) [reply src/components/assets/Account.vue]: Fixed it in this [commit](https://github.com/AstarNetwork/astar-apps/pull/956/commits/0b71f3a800b60d1bae2a14f328a21aaea62291e1)

File: src/components/assets/Account.vue (hunk 1 above)
Comment (ayumitk) [reply src/components/assets/Account.vue]: Fixed it in this [commit](https://github.com/AstarNetwork/astar-apps/pull/956/commits/0c5ca2f0a0b5e792786a55fe339f2ae21fcdbbe4)

File: src/components/assets/Assets.vue (hunk 2 above)
Comment (ayumitk) [reply src/components/assets/Assets.vue:159]: Let me work on this in another PR. I'll do it next week

File: src/components/assets/EvmNativeToken.vue (hunk 5 above)
Comment (ayumitk) [reply src/components/assets/EvmNativeToken.vue]: I've moved svg icon to Astar UI and replaced it in this [commit](https://github.com/AstarNetwork/astar-apps/pull/956/commits/ee35c19b4b8422c31510093c4f2735c7b75bf87e)

File: src/components/header/TroubleHelp.vue (hunk 6 above)
Comment (ayumitk) [reply src/components/header/TroubleHelp.vue]: I've moved svg icon to Astar UI and replaced it in this [commit](https://github.com/AstarNetwork/astar-apps/pull/956/commits/ee35c19b4b8422c31510093c4f2735c7b75bf87e)

**Comment by ayumitk** [comment]
> Let's move some icons to Astar-UI later.

@impelcrypto I've moved svg icon to Astar UI and replaced them in this [commit](https://github.com/AstarNetwork/astar-apps/pull/956/commits/ee35c19b4b8422c31510093c4f2735c7b75bf87e)

**Review by bobo-k2 — APPROVED** [review]
LGTM

**Pull request merged**